rtfs.go

```diff
@@ -102,6 +102,24 @@ func (im *IpfsManager) Pin(hash string) error {
 	return nil
 }
 
+// PinUpdate is used to update one pin to another, while making sure all objects
+// in the new pin are local, followed by removing the old pin.
+//
+// This is an optimized version of pinning the new content, and then removing the
+// old content.
+//
+// returns the new pin path
+func (im *IpfsManager) PinUpdate(fromPath, toPath string) (string, error) {
+	out, err := im.shell.PinUpdate(fromPath, toPath)
+	if err != nil {
+		return "", err
+	}
+	if len(out) == 0 {
+		return "", errors.New("failed to retrieve new pin paths")
+	}
+	return out["Pins"][1], nil
```

Comment:
we should check for length here too, since this might panic

Reply:
Good catch